### PR TITLE
Support prompting user for program arguments

### PR DESCRIPTION
**Pull request by testforstephen**
Implement the feature to support prompting user for program arguments: https://github.com/Microsoft/vscode-java-debug/issues/228

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,7 +31,8 @@
   "activationEvents": [
     "onLanguage:java",
     "onDebugInitialConfigurations",
-    "onDebugResolve:java"
+    "onDebugResolve:java",
+    "onCommand:vscJavaDebug.SpecifyProgramArgs"
```
Comment (yaohaizh) [inline package.json]: It is not necessary put vsc as the prefix since all the code base is working inside the VSCode scope.

File: package.json [hunk 2]
```diff
@@ -284,6 +288,20 @@
               "args": ""
             }
           },
+          {
+            "label": "Java: Launch Program with Args Prompt",
+            "description": "Add a new configuration for launching a java program, prompting first for program arguments.",
+            "body": {
+              "type": "java",
+              "name": "Debug (Launch) with Args Prompt",
```
Comment (yaohaizh) [inline package.json]: Same with above, should not use abbreviation in the description to the end user.

File: src/extension.ts [hunk 3]
```diff
@@ -55,3 +58,26 @@ export function deactivate() {
 function fetchUsageData() {
     return commands.executeJavaLanguageServerCommand(commands.JAVA_FETCH_USAGE_DATA);
 }
+
+function specifyProgramArguments(context: vscode.ExtensionContext): Thenable<string> {
```
Comment (andxu) [inline src/extension.ts]: add readme instructions.

**Comment by yaohaizh** [comment]
Is it really necessary to add a configuration generate for only a launch options?  It kindly boost the number of the configuration if we add the a config for each options?  Should a options inside the launch.json is enough?

**Comment by yaohaizh** [comment]
Also, please see how other languages handle this issue.

**Comment by testforstephen** [comment]
This is totally referring to `PowerShell`'s implementation.

File: package.json [hunk 4]
```diff
@@ -284,6 +288,20 @@
               "args": ""
             }
           },
+          {
+            "label": "Java: Launch Program with Args Prompt",
```
[the remark this answers is not in the document]
Comment (testforstephen) [reply package.json]: In launch.json, the prompting key name is `args`, using `Args Prompt` is suitable here.

File: README.md [hunk 5]
```diff
@@ -47,7 +47,7 @@ Please also check the documentation of [Language Support for Java by Red Hat](ht
 ### Launch
 
 - `mainClass` (required) - The main class of the program (fully qualified name, e.g. [mymodule/]com.xyz.MainClass).
-- `args` - The command line arguments passed to the program.
+- `args` - The command line arguments passed to the program. Support prompting for program arguments by using variable `"${command:SpecifyProgramArgs}"`.
```
Comment (akaroml) [inline README.md]: Use `"${command:SpecifyProgramArgs}"` to specify program arguments for each debug session.

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (akaroml) [reply package.json]: Java: Launch Program by Prompting Arguments

File: package.json [hunk 6]
```diff
@@ -284,6 +288,20 @@
...
+            "description": "Add a new configuration for launching a java program, prompting first for program arguments.",
```
Comment (akaroml) [inline package.json]: ...for launching a java program by prompting arguments.

File: package.json (hunk 2 above)
Comment (akaroml) [reply package.json]: Debug (Launch) by Prompting Arguments

File: README.md (hunk 5 above)
Comment (testforstephen) [reply README.md]: Feels like the new message doesn't reflect the prompting way to configure args. And not each debug session would take this way to configure args. This is an optional way. The `args` in launch.json still can be a hard-code constant string (previous design) or prompting.

How about `Use "${command:SpecifyProgramArgs}" to prompt user to specify program arguments dynamically.`?

File: package.json (hunk 4 above)
[the remark this answers is not in the document]
Comment (testforstephen) [reply package.json]: The sentence pattern fully follows the powershell extension. There it used `Launch ... with ...`. https://github.com/PowerShell/vscode-powershell/blob/master/package.json#L235

Not sure whether `Launch ... by ...` is better. I need look up in a dictionary.

File: README.md (hunk 5 above)
Comment (akaroml) [reply README.md]: Use `"${command:SpecifyProgramArgs}"` to prompt for program arguments.

**Pull request merged**